General/Glossary.md

```diff
@@ -2,38 +2,64 @@
 
 ## Candidates
 In the context of a method-call the candidates are the potential classes in the system that can receive the call. This list is computed from a static analysis of the source code.
 
+## Changes (file)
+The changesfile logs of all source code modifications (especially all thechanges you did while programming) since the [sources file](#sources) was generated. This facilitates a per method history for diffs or reverting. It means that even if you did not manage to save the image file on a crash or you just forgot, you can recover your changes from this file. A changes file is always coupled with a image file. They work in pair.
+
+> Note: Since Pharo 5 Pharo contains a project called Epicea implementing a new logging system. The long term goal of Epicea is to replace the changes file, but we are not yet at this stage.
+
 ## Class-side
 
+## Cog
+Cog is the name of the [virtual machine](#virtual-machine) used by Pharo. Cog is a virtual machine used by multiple programming languges such as Pharo, [Squeak](#squeak), Newspeak and Cuis.
+
 ## Context
 
 ## Debugger
 
 ## Dispatch
-Dispatch is a technique used in object-oriented programming to delegate behavior using polymorphism. The goal is to define a concept that will changed depending on the context. For example we can have an application with commands. When we need to execute the command, instead of creating a switch/case, each object will implément their own behavior on a method of the same name and we just need to call this method. 
+Dispatch is a technique used in object-oriented programming to delegate behavior using [polymorphism](#polymorphism). The goal is to define a concept that will changed depending on the context. For example we can have an application with commands. When we need to execute the command, instead of creating a switch/case, each object will implement their own behavior on a method of the same name and we just need to call this method. 
 
-## DoesNotUnderstand (DNU)
-This name is used to designate the error that arise when a message is sent to an object but this object does not understand it. It also happen that people use the "DNU" shortcut.
+## DNU
+See [DoesNotUnderstand](#doesnotunderstand).
+
+## DoesNotUnderstand
+This name is used to designate the error that arise when a [message](#message) is sent to an object but this object does not understand it. It also happen that people use the "DNU" shortcut.
 
 ## Iceberg
 
+## Image
+A Pharo image is a snapshot of the system memory of the Pharo system at a given moment. It is called image since it is a snapshot. This is the file where all objects are stored and as such it’s a cross plat-form format. An image file contains the live state of all objects of the system (including classes and compiled methods, since they are objects too) at a givenpoint. An image is a virtual object container. 
```

Comment:
```suggestion
A Pharo image is a snapshot of Pharo memory at a given moment.
This is the file where all objects are stored and as such it’s a cross platform format.
An image file contains the live state of all objects of the system (including classes and compiled methods, since they are objects too) at a givenpoint.
It can bee seen as a virtual object container.
```